ubyssey/events/models.py

```diff
@@ -91,6 +91,30 @@ def send_submitted_email(sender, instance, **kwargs):
         instance.is_submission_email = True
         instance.save()
 
+
+@receiver(pre_save, sender=Event)
+def format_event_ticket_url(sender, instance, **kwargs):
+    urls_formatted = ( ( has_protocol(instance.event_url) or not instance.event_url ) and
+                       ( has_protocol(instance.ticket_url) or not instance.ticket_url ) )
+    
+    """Add "http://" in front of (non-empty) event_url and ticket_url if protocol missing."""
+    if instance.is_submission and not urls_formatted:
```

Comment:
In this case, you don't have to check if the instance is a submission. I think it's safe to say that the URLs should be sanitized whenever the event is being saved.

Furthermore, I don't think the `urls_formatted` check is necessary. You should be able to just do this without the `if` statement:

```py
instance.event_url = format_url(instance.event_url)
instance.ticket_url = format_url(instance.ticket_url)
```
If the url is already formatted, the above code won't modify it.

You should also be able to omit `instance.save()` because the instance is about to be saved anyways.

Reply:
You're absolutely right... I'm not sure what I was doing wrong the other day because when I tested it without some sort of check, I had gotten an error about infinite recursion or something to that effect. But I tried it again right now and everything's fine!

Reply:
Awesome!

It's likely that the call to `instance.save()` was creating a loop by triggering the `pre_save` hook again.